mempool/src/shared_mempool.rs

```diff
@@ -157,13 +164,16 @@ async fn sync_with_peers<'a>(
 
             if !transactions.is_empty() {
                 OP_COUNTERS.inc_by("smp.sync_with_peers", transactions.len());
-                let mut msg = MempoolSyncMsg::default();
-                msg.peer_id = peer_id.into();
-                msg.transactions = transactions
+                let mut req = SubmitTransactionsRequest::default();
```

Comment:
So this diff moves us into intermediate state where main broadcast routine still operates based on direct send and we just adding functionality to listen to potential rpc calls. 
Is that correct? That moves us into intermediate state and there's a bit of new code duplication(transaction validation, etc).
Is a plan to build this feature as series of stacked diffs and ship all at once?

Reply:
right - I'll have a stack of dependent PR's, and this is the first in the stack